packages/p2p/src/components/my-profile/payment-methods/add-payment-method/cancel-add-payment-method-modal.jsx

```diff
@@ -30,7 +35,15 @@ const CancelAddPaymentMethodModal = () => {
                         my_profile_store.setIsCancelAddPaymentMethodModalOpen(false);
                         my_profile_store.setSelectedPaymentMethod('');
                         my_profile_store.setSelectedPaymentMethodDisplayName('');
-                        my_profile_store.setShouldShowAddPaymentMethodForm(false);
+                        // ensuring the previous modal is closed before opening the new modal
+                        const close_modal = setInterval(() => {
+                            if (!my_profile_store.is_modal_open) {
+                                buy_sell_store.setShouldShowPopup(true);
+                                clearInterval(close_modal);
+                            }
+                        });
```

Comment:
@likhith-deriv why do we need `setInterval`? We should track the opened modal properly.
Using `setInterval` for this kind of checking is highly discouraged. Please use promise if needed.

Reply:
I will replace this with when() of mobx